jax/_src/numpy/util.py

```diff
@@ -311,6 +311,20 @@ def promote_dtypes_complex(*args: ArrayLike) -> list[Array]:
           for x in args]
 
 
+def promote_dtypes_integral_default(*args: ArrayLike) -> list[Array]:
+  """Convenience function to apply default promotion to integral accumulators.
+
+  Promotes arguments to their corresponding default integral type, or returns
+  the arguments unchanged."""
+  def _promote(x: ArrayLike) -> Array:
+    x = lax.asarray(x)
+    kind = x.dtype.kind
+    if kind in {'i', 'u'}:
+      return x.astype(dtypes._default_types[kind])
```

Comment:
Is this correct? What if `x.dtype` is `int64` and `default_types[kind]` is `int32` – I don't think we want to demote in that case.